pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile.go

```diff
@@ -49,7 +49,7 @@ func (c *Controller) reconcile(ctx context.Context, ws *tenancyv1alpha1.ClusterW
 			listShards: c.rootWorkspaceShardLister.List,
 		},
 		&phaseReconciler{
-			getShardWithQuorum: c.kcpClient.Cluster(tenancyv1alpha1.RootCluster).TenancyV1alpha1().ClusterWorkspaceShards().Get,
```

Comment:
I agree this is a bit of a backwards change, but it does get us away from maintaining a fork of client-gen...

What if we did something like this?
```go
getShardWithQuorum: func(ctx context.Context, name string, options metav1.GetOptions) (*tenancyv1alpha1.ClusterWorkspaceShard, error) {
    rootCtx := logicalcluster.WithCluster(ctx, tenancyv1alpha1.RootCluster)
    return c.kcpClient.TenancyV1alpha1().ClusterWorkspaceShards().Get(rootCtx, name, options)
}
```

This doesn't fix it, but it does replicate to a degree what we had before - in the `getShardWithQuorum` declaration, you are setting the scope, as opposed to doing it in the `phaseReconciler` where `getShardWithQuorum` is invoked.

The only other option I can think of (that doesn't involve generating a clientset with `.Cluster()`) would be something like a "KCPClientFactory" that could hand out scoped clients, but this would probably be less efficient because they wouldn't share the same underlying http.Client by default, unless we did some extra work there.